doc/VTGateV3Features.md

```diff
@@ -0,0 +1,86 @@
+# VTGate V3 features
+
+## Overview
+Historically, Vitess was built from underneath YouTube. This required us to take an iterative approach, which resulted in many versions:
+* V0: This version had no VTGate. The application was expected to fetch the sharding info and tablet locations from a 'toposerver' and use it to talk directly to the tablet servers.
+* V1: This was the first version of VTGate. In this version, the app only needed to know the number of shards, and how to map the sharding key to the correct shard. The rest was done by VTGate. In this version, the app was still exposed to resharding events.
+* V2: In this version, the keyspace id was required instead of the shard. This allowed the app to be agnostic of the number of shards.
+
+With V3, the app does not need to specify any routing info. It just sends the query to VTGate as if it's a single database. Apart from simplifying the API, there are some additional benefits:
+* Database compliant drivers can be built for each client language. This will also allow for integration with third party tools that work with such drivers.
+* V3 can aspire to satisfy the full SQL syntax. This means that it will be able to perform cross-shard joins, aggregations and sorting.
+* Easier migration: an application that was written to use a single database can be trivially changed to use Vitess, and then the database can be scaled from underneath without changing much of the app.
+
+## Feature set
+The [V3 design](https://github.com/youtube/vitess/blob/master/doc/VTGateV3.md) is quite elaborate. If necessary, it will allow you to plug in custom indexes and sharding schemes. However, it comes equipped with some pre-cooked recipes that satisfy the immediate needs of the real-world:
+
+### Knowing where tables are
```

Comment:
this section should probably to go the end of the feature set section? Most people are interested in sharding horizontally first and not vertically? Therefore, I would talk about horizontal sharding features first and then vertical sharding.